src/wrapper.jl

```diff
@@ -106,22 +214,952 @@ const FT_PURGE_TX = 2
+
+See D2XX Programmer's Guide (FT_000071) for more information.
+
+# Example
+```julia-repl
+julia> numdevs = FT_CreateDeviceInfoList()
+0x00000004
+
+julia> "Number of devices is \$numdevs"
+"Number of devices is 4"
+```
+"""
+function FT_CreateDeviceInfoList()
+  lpdwNumDevs = Ref{DWORD}(0)
+  status = ccall(cfunc[:FT_CreateDeviceInfoList], cdecl, FT_STATUS, 
+                 (Ref{DWORD},),
+                 lpdwNumDevs)
+  FT_STATUS_ENUM(status) == FT_OK || throw(FT_STATUS_ENUM(status))
+  lpdwNumDevs[]
+end
+
+
+
+"""
+    FT_GetDeviceInfoList(lpdwNumDevs)
+
+Wrapper for D2XX library function `FT_GetDeviceInfoList`. 
+
+See D2XX Programmer's Guide (FT_000071) for more information.
+
+# Arguments
+ - `lpdwNumDevs`: The number of devices.
+
+# Example
+```julia-repl
+julia> numdevs = FT_CreateDeviceInfoList()
+0x00000004
+
+julia> devinfolist, numdevs = FT_GetDeviceInfoList(numdevs);
+
+julia> numdevs
+0x00000004
+
+julia> using LibFTD2XX.Util # for ntuple2string
+
+julia> ntuple2string(devinfolist[1].description)
+"USB <-> Serial Converter D"
+
+julia> devinfolist[1].fthandle_ptr
+Ptr{Nothing} @0x0000000000000000
+
+julia> devinfolist[1].locid
+0x00000000
+
+julia> devinfolist[1].typ
+0x00000007
+
+julia> devinfolist[1].flags
+0x00000002
+
+julia> devinfolist[1].id
+0x04036011
+
+julia> ntuple2string(devinfolist[1].serialnumber)
+"FT3AD2HCD"
+
+```
+"""
+function FT_GetDeviceInfoList(lpdwNumDevs)
+  pDest =  @compat Vector{FT_DEVICE_LIST_INFO_NODE}(undef, lpdwNumDevs)
+  status = ccall(cfunc[:FT_GetDeviceInfoList], cdecl, FT_STATUS, 
+                 (Ref{FT_DEVICE_LIST_INFO_NODE}, Ref{DWORD}),
+                  pDest,                         Ref{DWORD}(lpdwNumDevs))
+  pDest, lpdwNumDevs
+end
+
+
+
+"""
+    FT_GetDeviceInfoDetail(dwIndex)
+
+Wrapper for D2XX library function `FT_GetDeviceInfoDetail`. 
+
+See D2XX Programmer's Guide (FT_000071) for more information.
+
+# Arguments
+ - `dwIndex`: Index of entry in the device info list.
+
+# Example
+```julia-repl
+julia> numdevs = FT_CreateDeviceInfoList()
+0x00000004
+
+julia> idx, flags, typ, id, locid, serialnumber, description, fthandle = FT_GetDeviceInfoDetail(0) # zero indexed
+(0, 0x00000002, 0x00000007, 0x04036011, 0x00000000, "FT3AD2HCD", "USB <-> Serial Converter D", FT_HANDLE(Ptr{Nothing} @0x0000000000000000))
+```
+"""
+function FT_GetDeviceInfoDetail(dwIndex)
+  lpdwFlags, lpdwType  = Ref{DWORD}(), Ref{DWORD}()
+  lpdwID,    lpdwLocId = Ref{DWORD}(), Ref{DWORD}()
+  pcSerialNumber = pointer(Vector{Cchar}(undef, 16))
+  pcDescription  = pointer(Vector{Cchar}(undef, 64))
+  ftHandle = FT_HANDLE()
+  
+  status = ccall(cfunc[:FT_GetDeviceInfoDetail], cdecl, FT_STATUS, 
+  (DWORD,   Ref{DWORD}, Ref{DWORD}, Ref{DWORD}, Ref{DWORD}, Cstring,        Cstring,       Ref{FT_HANDLE}),
+   dwIndex, lpdwFlags,  lpdwType,   lpdwID,     lpdwLocId,  pcSerialNumber, pcDescription, ftHandle)
+  
+  FT_STATUS_ENUM(status) == FT_OK || throw(FT_STATUS_ENUM(status))
```

Comment:
What exactly are you throwing here and elsewhere?

Reply:
An `FT_STATUS_ENUM`. It should actually be a special exception type. This is noted as a problem in issues in main PR comment:

>  - `.Wrapper` functions which fail throw `FT_STATUS_ENUM`s instead of exceptions. This should be replaced with a suitable exception type.

Reply:
I will open issues for everything in the issue list of the PR once this is approved